lib/assert.js

```diff
@@ -320,6 +320,18 @@ function wrapAssertions(callbacks) {
 				return;
 			}
 
+			try {
+				if (isPromise(fn().catch(_ => {}))) {
```

Comment:
You should move this condition down to after the `retval = fn()` line, and check if `isPromise(retval)`. Currently this calls `fn` twice, which is not what we want.

I'd only hook up the no-op `catch()` once we know `retval` is a promise. There's a `noop` function already declared that you can use.